third_party/2and3/mypy_extensions.pyi

```diff
@@ -1,6 +1,11 @@
-from typing import Dict, Type, TypeVar
+from typing import Dict, Type, TypeVar, Union
 
 T = TypeVar('T')
 
 
 def TypedDict(typename: str, fields: Dict[str, Type[T]]) -> Type[dict]: ...
+
+# Return type that indicates a function does not return.
+# This type is equivalent to the None type, but the no-op Union is necessary to
+# distinguish the None type from the None value.
+NoReturn = Union[None]
```

Comment:
The intent is to not disrupt typecheckers that don't yet support this (as it's not yet part of PEP 484).